### PR TITLE
Fix SafeMath test file

### LINKED ISSUE #854: SafeMath.test.js Bug
## 🎉 Description

Overflow tests are using integers which are way above Javascript's MAX_SAFE_INTEGER.

- [x] 🐛 This is a bug report.
- [ ] 📈 This is a feature request.

## 💻 Environment

Any platform.

## 📝 Details

In the overflow tests for SafeMath.sol, you are using integers which are way above Javascript's MAX_SAFE_INTEGER. Whether or not these tests work, I think is by mere coincidence (kind of *undefined behavior*, if this terminology applies for Javascript).

## 🔢 Code To Reproduce Issue

```
let x = 115792089237316195423570985008687907853269984665640564039457584007913129639935;
let y = 115792089237316195423570985008687907853269984665640564039457584007913129639936;
alert(x < y);
```

UPDATE:

In fact, this test passes by cheating.

It checks for `revert`, where the actual error to be expected is `invalid opcode`.
This is because the `SafeMath.sol` contract uses `assert`s rather than `require`s.

So the fix should include:

1. Replace each one of those large integers with `web3.toBigNumber("...")`.
2. Replace each `assertRevert` with `assertJump`.

Alternatively for the 1st bullet (a more appropriate fix IMO), you can simply use:

    let a = web3.toBigNumber(2).pow(256).minus(1)

BTW, I haven't been quite able to determine why a `revert` exception is thrown here to begin with.

---

Here is the exact fix that I have used:

```
  it('should throw an error on addition overflow', async function () {
    let a = web3.toBigNumber(2).pow(256).minus(1);
    let b = 1;
    try {
      await safeMath.add(a, b);
      assert.fail('should have thrown before');
    } catch (error) {
      assertJump(error);
    }
  });

  it('should throw an error on multiplication overflow', async function () {
    let a = web3.toBigNumber(2).pow(256).div(2);
    let b = 2;
    try {
      await safeMath.multiply(a, b);
      assert.fail('should have thrown before');
    } catch (error) {
      assertJump(error);
    }
  });
```

### PR DESCRIPTION
**Pull request by facuspagnuolo**
Fixes #854 
@frangio @Shrugs please review

### CODE REVIEW

File: test/helpers/assertJump.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
-module.exports = function (error) {
-  assert.isAbove(error.message.search('invalid opcode'), -1, 'Invalid opcode error must be returned');
+export default async promise => {
+  try {
+    await promise;
+    assert.fail('Expected invalid opcode not received');
```
Comment (Shrugs) [inline test/helpers/assertJump.js]: can we change it to `"invalid opcode" not received"` to be more clear?

File: test/math/SafeMath.test.js [hunk 2]
```diff
@@ -1,60 +1,91 @@
...
+const SafeMathMock = artifacts.require('SafeMathMock');
 
-contract('SafeMath', function (accounts) {
-  let safeMath;
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('SafeMath', () => {
+  const MAX_UINT = new BigNumber('115792089237316195423570985008687907853269984665640564039457584007913129639935');
```
Comment (Shrugs) [inline test/math/SafeMath.test.js]: Do we not want to do `new BigNumber(1).pow(256).sub(1)` or whatever that was to get the max number?

File: test/math/SafeMath.test.js [hunk 3]
```diff
@@ -1,60 +1,91 @@
...
-    await safeMath.add(a, b);
-    let result = await safeMath.result();
+      const result = await this.safeMath.add(a, b);
+      result.should.be.bignumber.equal(a.plus(b));
+    });
 
-    assert.equal(result, a + b);
+    it('throws an error on addition overflow', async function () {
+      const a = MAX_UINT;
+      const b = new BigNumber(1);
```
Comment (Shrugs) [inline test/math/SafeMath.test.js]: can we test the case of `+1` and `+2` here? I feel like an overflow to 0, which obviously included, might be a special case worth checking?

**Comment by facuspagnuolo** [comment]
I'll address those comments in https://github.com/OpenZeppelin/zeppelin-solidity/pull/835 :)

**Pull request merged**